### ISSUE TITLE
Incremental import of 2016 netfile data

### ISSUE THREAD

**Issue by adborden** [caciviclab/disclosure-backend#125]
Two parts, first, lets add 2016 to the list of years that we pull from netfile.

The second part, we now try to be a little smarter about only importing data we haven't yet imported. The current implementation works on a per-file basis and therefore imports the entire year at a time. This won't work for 2016, where data will be rolling in based on the filing schedule. Having a data file for 2016 does not mean there's no more data to pull in.

**Comment by adborden** [comment]
Some exploration... the export endpoint `/public/campaign/export/cal201/transaction/year` isn't returning any 2016 data yet but the transaction search endpoint `/public/campaign/search/transaction/query` does. The transaction search endpoint also allows us to specify a `dateStart` parameter which we can use to pull all the transactions since our last fetch by specifying the most recent transaction date in the DB.

```
$ curl 'https://netfile.com:443/Connect2/api/public/campaign/search/transaction/query' -X POST -d DateStart=2016-01-01T00:00:00Z -d transactionType=0 -H 'Accept: application/json' # yields 1 transaction
```

Open questions:
1. Is querying by `dateStart` (the transaction date) good enough or will we miss transactions because they are filed way after the actual transaction date. i.e. do we really need to be looking at filings here?
2. When will the export data endpoint have data?

**Comment by adborden** [comment]
cc @bcipolli @andrell81 questions for netfile ^^

**Comment by tdooner** [comment]
Also, I suspect we'll need to be careful regarding amendments. I believe amended filings usually contain all the original rows but with a larger value for the `Report_Num` field.

**Comment by adborden** [comment]
Yeah, I think the amendments will be handled by the data pipeline, this story is about fetching/importing raw transactions into `netfile_raw`.

**Comment by bcipolli** [comment]
@adborden Thanks for the info! We should hop to this ASAP. I will take a quick look now.

**Comment by bcipolli** [comment]
Note that with #185, we don't necessarily have to skip pulling old data. Indeed, it may be useful to have some date overlap when pulling to start, to be robust against any possible Netfile weirdness in reporting.

**Comment by bcipolli** [comment]
Sadly, the schema difference between the `ByYear` and `Query` endpoints greatly differ. I will try to get in touch with our Netfile contact to discuss.

**Comment by bcipolli** [comment]
I am satisfied for the current RC with our current workflow: force-redownload the 2016 data (slow), then skip rows of transactions previously loaded (fast). Will punt this to the next RC, to try and find a download endpoint that only has new data.
